### PR TITLE
fix(rpc): `masternode outputs` should work correctly with multiple outputs created via a single tx

### PR DESCRIPTION
**Pull request by UdjinM6**
Currently it only shows the last one since it's an object with the same key (txid) for every outpoint.

kudos to @kxcd for discovering this 👍

### CODE REVIEW

File: test/functional/rpc_masternode.py [hunk 1]
```diff
@@ -66,6 +66,13 @@ def run_test(self):
             assert_equal(gbt_masternode[i]["script"], payments_masternode["payees"][i]["script"])
             assert_equal(gbt_masternode[i]["amount"], payments_masternode["payees"][i]["amount"])
 
+        self.log.info("test that `masternode outputs` show correct list")
+        addr1 = self.nodes[0].getnewaddress()
+        addr2 = self.nodes[0].getnewaddress()
+        self.nodes[0].sendmany('', {addr1: 1000, addr2: 1000})
+        self.nodes[0].generate(1)
+        # we have 3 masternodes that are running already and 2 new outputs we just created
+        assert_equal(len(self.nodes[0].masternode("outputs")), 5)
```
Comment (PastaPastaPasta) [inline test/functional/rpc_masternode.py:75]: Thank you for me not needing to ask for this!!! :)

**Review by PastaPastaPasta — DISMISSED** [review]
Looks like the format of the rpc output might be different.. Should this be documented? and is this okay to merge right now for inclusion in 18.1 or should this wait for 19?

**Comment by UdjinM6** [comment]
> Looks like the format of the rpc output might be different.. 

Yes, it changes format from `{"txid":"nout", ...}` to `["txid-nout", ...]`

> Should this be documented?

Ping @thephez 

> and is this okay to merge right now for inclusion in 18.1 or should this wait for 19?

It's a non-critical rpc (no potential funds loss or other serious issues - some scripts might fail if they use it but should be trivial to do), imo it's fine to merge in 18.1. But I won't insist :)

**Comment by kxcd** [comment]
But I won't insist :)
I will, please push this to v18.1 it is impacting whales because I needed the RPC to work and without it and a good explorer I had no idea how the outputs were ordered and which were valid for my masternode(s).

**Comment by PastaPastaPasta** [comment]
> ```
> But I won't insist :)
> ```
> 
> 
>      
>   
> 
> I will, please push this to v18.1 it is impacting whales because I needed the RPC to work and without it and a good explorer I had no idea how the outputs were ordered and which were valid for my masternode(s).

@kxcd This was gonna be pushed to 18.1 no matter what, the question was should it be pushed to v19

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge for v18.1

**Pull request merged**